src/hpx_init.cpp

```diff
@@ -647,6 +639,13 @@ namespace hpx
                 std::unique_ptr<hpx::runtime> rt(
                     new runtime_type(cms.rtcfg_));
 
+                // check whether HPX should be exited at this point
+                // (if the program options contain --hpx:help or --hpx:version)
+                if (rp.terminate_after_parse())
+                {
+                    return result;
```

Comment:
Why have you remove conditionally setting of the result value to zero?

Reply:
Because it's always zero at this point in time as far as I can see.

Reply:
You're right. The code as it is right now leaves result unchanged after it was initialized. This is a leftover problem from our move to the `resource_partitioner`, though. I'll try to come up with a patch against this PR to fix things.